warehouse/upload.go

```diff
@@ -1067,10 +1068,25 @@ func (job *UploadJobT) loadTable(tName string) (alteredSchema bool, err error) {
 		job.recordTableLoad(tName, numEvents)
 	}
 
-	if columnThreshold, ok := columnCountThresholds[job.warehouse.Type]; ok {
-		columnCount := len(job.schemaHandle.schemaInWarehouse[tName])
-		if columnCount > columnThreshold {
-			job.counterStat(`warehouse_load_table_column_count`, tag{name: "tableName", value: strings.ToLower(tName)}).Count(columnCount)
+	job.columnCountStat(tName)
+
+	return
+}
+
+func (job *UploadJobT) columnCountStat(tableName string) {
+	if columnCountLimit, ok := columnCountLimitMap[job.warehouse.Type]; ok {
+		currentColumnsCount := len(job.schemaHandle.schemaInWarehouse[tableName])
+		if currentColumnsCount > int(float64(columnCountLimit)*columnCountLimitThreshold) {
+			tags := []tag{
+				{
+					name: "tableName", value: strings.ToLower(tableName),
+				},
+				{
+					name: "columnCountLimit", value: strconv.Itoa(columnCountLimit),
+				},
+			}
+
+			job.counterStat(`warehouse_load_table_column_count`, tags...).Count(currentColumnsCount)
 		}
 	}
```

Comment:
```suggestion
	if columnCountLimit, ok := columnCountLimitMap[job.warehouse.Type]; !ok {
	    return
	}
	currentColumnCount := len(job.schemaHandle.schemaInWarehouse[tableName])
	if currentColumnsCount > int(float64(columnCountLimit)*columnCountLimitThreshold) {
		tags := []tag{
			{
				name: "tableName", value: strings.ToLower(tableName),
			},
			{
				name: "columnCountLimit", value: strconv.Itoa(columnCountLimit),
			},
		}

		job.counterStat(`warehouse_load_table_column_count`,
		tags...).Count(currentColumnsCount)
	}
```

Reply:
Just returning early to save on the indentation. Wdyt ?

Reply:
Done.